src/main/java/store/dto/SaleDTO.java

```diff
@@ -24,8 +26,11 @@
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
     private LocalDate dateOff;
 
+    @Min(value = 0, message = "percent must not be negative")
```

Comment:
I think that condition from product price doesn't fully fir here, cause sale percent must  be  higher than 1 and lower than 100.